streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java

```diff
@@ -416,6 +429,194 @@ public int compare(final KeyValue<String, Long> o1, final KeyValue<String, Long>
 
     }
 
+    @Test
+    public void shouldCountSessionWindows() throws Exception {
+        final long sessionGap = 5 * 60 * 1000L;
+        final long maintainMillis = sessionGap * 3;
+
+        final long t1 = mockTime.milliseconds() - TimeUnit.MILLISECONDS.convert(1, TimeUnit.HOURS);
+        final List<KeyValue<String, String>> t1Messages = Arrays.asList(new KeyValue<>("bob", "start"),
+                                                                        new KeyValue<>("penny", "start"),
+                                                                        new KeyValue<>("jo", "pause"),
+                                                                        new KeyValue<>("emily", "pause"));
+
+        IntegrationTestUtils.produceKeyValuesSynchronouslyWithTimestamp(
+                userSessionsStream,
+                t1Messages,
+                TestUtils.producerConfig(
+                        CLUSTER.bootstrapServers(),
+                        StringSerializer.class,
+                        StringSerializer.class,
+                        new Properties()),
+                t1);
+
+        final long t2 = t1 + (sessionGap / 2);
+        IntegrationTestUtils.produceKeyValuesSynchronouslyWithTimestamp(
+                userSessionsStream,
+                Collections.singletonList(
+                        new KeyValue<>("emily", "resume")
+                ),
+                TestUtils.producerConfig(
+                        CLUSTER.bootstrapServers(),
+                        StringSerializer.class,
+                        StringSerializer.class,
+                        new Properties()),
+                t2);
+        final long t3 = t1 + sessionGap;
```

Comment:
`final long t3 = t1 + sessionGap + 1`; -- otherwise I would assume that the sessions get merged.